### ISSUE TITLE
To-do tasks for the new installer

### ISSUE THREAD

**Issue by sammilucia** [snowie2000/mactype#443]
When the new installer (currently in testing) is run for the first time it should:
1. Show a welcome message
1. Clean up conflicting files such as EasyHK*.dll from various locations
1. Initiate cleanup of the old installer to completely remove remnants of the old installation AND the old installer (files and registry keys)
1. Leave custom profiles intact
1. Show a welcome message
1. (Maybe) show a finished message
1. (Optional) Launch MacWiz with elevated privileges on closing the installer wizard
1. (Optional) Create Desktop shortcuts for all MacType *.exes
1. Properly set Compatibility of the MacWiz.exe icons as Run as Admin

**Comment by sammilucia** [comment]
Still using Inno Setup, I believe it can do everything we need pretty easily. In-dev installer is a bit of a mess while I learn it but I'll go thru and clean that up shortly.

**Comment by sammilucia** [comment]
@snowie2000 for #358 the new installer supporting regional decimals... I can have two sets of INIs one for comma decimals and one for period decimals. Is this wise though? It increases our work, and I don't know that people expect regional settings to be supported in ini files and will cause support issues... Should we standardise on one?

**Comment by snowie2000** [comment]
@sammilucia I think, even those people who set their decimal separator to comma are actually used to use period as their daily separators.
As I have solved that locale problem, I don't think we need another whole set of profiles with the only difference of decimal separators. Users should be happy with period as the separator as it is commonly used worldwide.

**Comment by sammilucia** [comment]
- Added pre-install code
  - call previous uninstaller
  - forceful cleanup
  - structure for manually stopping services / de-registering DLLs, killing mactray.exe, etc.
- Fixed duplicate file copy
- Structure for logging
- Structure for error handling

**Comment by sammilucia** [comment]
i think it's ready to go. i need people who can test it on a 32 bit Windows

bugfixes:
- Fixed old MacType uninstaller was no longer being invoked
- Added warnings when (un)installing to backup profiles first

**Comment by snowie2000** [comment]
Okay, I think  We can release a new beta tomorrow or next weeki, probably we can have a chance to test them together. 
Also, I have another question that does your installer come with an updater.

**Comment by sammilucia** [comment]
Yep, ready when you are. I'm suspicious 32 bit installs will have problems ... I might need to build a VM if there's no one to help test 😜

Yes there's a few options by the looks...

I think for maintainability since we all have other commitments, we should keep setup.exe simple to start. the simple way is to let Inno Setup handle the updates... it's very easy to create a new package now the groundwork is done.

1. use the existing updater.exe to check e.g. https://mactype.net/updater.txt, download new setup.exe when there is one, and execute as `/SP- /silent /noicons "/dir=expand:{pf}\setup.exe"` ... Inno Setup will update the existing install, handle restarts, etc... 

2 for a fresh install Inno Setup can do this check itself (and check the same updater.txt), and download and run the newer setup.exe payload as part of initial install.

There isn't a streaming/auto-updater as such ... if we were to do that i think using the built in VS deployment tools is probably smoothest. This can be a good option but takes some control away from users which I think has been important for MacType
